CMakeLists.txt

```diff
@@ -128,7 +128,7 @@ if (BUILD_STATIC_RELEASE)
   # Set the static variable
   set(P4C_STATIC_BUILD STATIC)
   # Do not bring in dynamic libstdcc and libgcc
-  set(CMAKE_EXE_LINKER_FLAGS "-static -static-libgcc -static-libstdc++ -Wl,-z,muldefs")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static-libgcc -static-libstdc++ -Wl,-z,muldefs")
```

Comment:
Linking protoc when using -static causes a series of problems. It seems to be a bad idea to use -static in general, so let's just fix this call.

Reply:
But this option is under `BUILD_STATIC_RELEASE`. It seems wrong to me to have this build non-static binary. If `BUILD_STATIC_RELEASE` is broken, we should just remove it. Arguably it is broken on most linux distros as glibc should not be linked statically (and often there is no corresponding `.a`). Therefore, `BUILD_STATIC_RELASE` would be useful only on distros like Alpine (which uses MUSL instead of GLIBC).

I suggest you just don't enable `BUILD_STATIC_RELEASE`. If you have need for semi-static (i.e. glibc dynamic, rest static) build, that should be handled by a new CMake option in a new PR.

Reply:
We a CI test that causes problems, so it is a little tricky to make this work. `-static` not only causes problems for Protobuf but also the [libgc](https://github.com/p4lang/p4c/pull/3930) build.

Let me experiment a bit with this. I am able to build locally with clang+lld. So maybe that is one way to ensure we can at least generate some form of static release. 

With bf-p4c, we did not even use the `-static` iirc.

Reply:
Yep, that is why I suggested actually removing static as a possibility. If we add semi-static instead, it would not be hard if anyone needed full-static to set it from it (basically just setting `CMAKE_EXE_LINKER_FLAGS` to `-static`). We probably have no way of knowing if static build is used by anyone (unless they show up here).

Reply:
So what is the best course of action here? Throw a warning when `BUILD_STATIC_RELEASE` is enabled? Or rename BUILD_STATIC_RELEASE?

Reply:
I would suggest renaming it. This seems the cleanest to me and least risk of breaking someones workflow. Maybe something like `BUILD_SEMI_STATIC_RELEASE`, although it does not explain what it means to be semi-static. But I cannot think of a short way to express "static except for basic OS dependencies like glibc".